exercises/react/tests/react.rs

```diff
@@ -101,32 +101,29 @@ fn error_setting_a_compute_cell() {
 
 use std::cell::RefCell;
 struct Expector {
-    have_value: RefCell<bool>,
-    value: RefCell<isize>,
+    value: RefCell<Option<isize>>,
 }
 
 impl Expector {
     fn new() -> Self {
         Expector {
-            have_value: RefCell::new(false),
-            value: RefCell::new(0),
+            value: RefCell::new(None),
         }
     }
 
     fn expect_to_have_been_called_with(&self, v: isize) {
-        assert!(*self.have_value.borrow(), "Callback was not called, but should have been");
-        assert_eq!(*self.value.borrow(), v, "Callback was called with incorrect value");
-        *self.have_value.borrow_mut() = false;
+        assert_ne!(*self.value.borrow(), None, "Callback was not called, but should have been");
+        assert_eq!(*self.value.borrow(), Some(v), "Callback was called with incorrect value");
+        *self.value.borrow_mut() = None;
     }
 
     fn expect_not_to_have_been_called(&self) {
-        assert!(!*self.have_value.borrow(), "Callback was called, but should not have been");
+        assert_eq!(*self.value.borrow(), None, "Callback was called, but should not have been");
     }
 
     fn callback_called(&self, v: isize) {
-        assert!(!*self.have_value.borrow(), "Callback was called too many times");
-        *self.value.borrow_mut() = v;
-        *self.have_value.borrow_mut() = true;
+        assert_eq!(*self.value.borrow(), None, "Callback was called too many times; can't be called with {}", v);
```

Comment:
also possible use of https://doc.rust-lang.org/std/cell/struct.RefCell.html#method.replace

Reply:
Agreed.